packages/block-library/src/block/index.php

```diff
@@ -78,13 +78,48 @@ function render_block_core_block( $attributes ) {
 	 * filter so that it is available when a pattern's inner blocks are
 	 * rendering via do_blocks given it only receives the inner content.
 	 */
-	$has_pattern_overrides = isset( $attributes['content'] );
+	$has_pattern_overrides = isset( $attributes['content'] ) && null !== get_block_bindings_source( 'core/pattern-overrides' );
 	if ( $has_pattern_overrides ) {
 		$filter_block_context = static function ( $context ) use ( $attributes ) {
 			$context['pattern/overrides'] = $attributes['content'];
 			return $context;
 		};
 		add_filter( 'render_block_context', $filter_block_context, 1 );
+
+		// Add bindings for the default pattern overrides source.
+		$apply_default_pattern_overrides_bindings = static function ( $parsed_block ) {
+			$supported_block_attrs = array(
+				'core/paragraph' => array( 'content' ),
+				'core/heading'   => array( 'content' ),
+				'core/image'     => array( 'id', 'url', 'title', 'alt' ),
+				'core/button'    => array( 'url', 'text', 'linkTarget', 'rel' ),
+			);
+
+			if (
+				// Return early if the block isn't one of the supported block types,
+				! isset( $supported_block_attrs[ $parsed_block['blockName'] ] ) ||
+				// or doesn't have a name,
+				! isset( $parsed_block['attrs']['metadata']['name'] ) ||
+				// or doesn't have the default binding.
+				empty( $parsed_block['attrs']['metadata']['bindings']['__default'] ) ||
+				// or the default binding isn't the pattern overrides source.
+				'core/pattern-overrides' !== $parsed_block['attrs']['metadata']['bindings']['__default']['source']
+			) {
+				return $parsed_block;
+			}
+
+			$bindings = array();
+			foreach ( $supported_block_attrs[ $parsed_block['blockName'] ] as $attribute_name ) {
+				$bindings[ $attribute_name ] = array( 'source' => 'core/pattern-overrides' );
+			}
+			$parsed_block['attrs']['metadata']['bindings'] = array_merge(
+				$bindings,
+				$parsed_block['attrs']['metadata']['bindings']
+			);
+
+			return $parsed_block;
+		};
+		add_filter( 'render_block_data', $apply_default_pattern_overrides_bindings, 10, 1 );
```

Comment:
I don't know about the implications of using a filter instead of changing [the bindings process](https://github.com/WordPress/wordpress-develop/blob/7a64d4044d31546602bd90cd4eda0add6f7733bd/src/wp-includes/class-wp-block.php#L242-L287) directly, as we are doing in the editor.

@gziolo Any idea if this could cause any issues?

Reply:
Updating the linked private method would be the preferred approach when integrating in WordPress Core. However that might be really difficult to achieve in the near perspective if we want to test it in the Gutenberg plugin.

By the way, the filter should get removed after executing `do_blocks`, otherwise, if there are several Synced Patterns on the page, the same filter will get added whenever a block renders. In effect 10 blocks rendered, the same filter registered and executed 10 times.

Reply:
Agree, I'd prefer it if there were some symmetry between. I'll look into it now, it would make the PHP backport easier if we decide to merge this for WordPress 6.6.